plugin/locationpicker.py

```diff
@@ -76,6 +77,9 @@ def _unpack(self, index: int) -> Tuple[Optional[Session], Union[Location, Locati
         return self._weaksession(), location, uri, position
 
     def _select_entry(self, index: int) -> None:
+        if self._view.is_valid() and not self._side_by_side:
+            self._view.set_viewport_position(self._view_states[1])
+            self._view.run_command('lsp_selection_set', {'regions': self._view_states[0]})
```

Comment:
Would it not make sense to do it only when index is `-1`? Otherwise it seems wasteful to run it if we're gonna override it below anyway. And it could cause some visual issues in theory (although I don't see any when testing here).

Reply:
I had tested this yesterday and the problem also exists in some cases when you select a valid entry. It happens when there is an entry with a location from the current file, and another entry with a location in different file. If you first highlight the location from the same file, it will scroll the viewport and update the selection, and if you then select the location in the other file, it will be opened in a new tab. But then the original file is still scrolled to the symbol position and the original selection is lost.